tests/integration/test_interface.py

```diff
@@ -300,6 +302,164 @@ def test_dataset_filter_by_year_with_full_dates(mocker, dataset_name: str):
+    """Test that dataset generation filtered by state noises data differently with different seeds"""
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir, seed=0, state=STATE)
+    noised_data_same_seed = noising_function(source=tmpdir, seed=0, state=STATE)
+    noised_data_different_seed = noising_function(source=tmpdir, seed=1, state=STATE)
+
+    assert not data.equals(noised_data)
+    assert noised_data.equals(noised_data_same_seed)
+    assert not noised_data.equals(noised_data_different_seed)
+
+
+@pytest.mark.parametrize(
+    "dataset_name",
+    [
+        DATASETS.census.name,
+        DATASETS.acs.name,
+        DATASETS.cps.name,
+        DATASETS.tax_w2_1099.name,
+        DATASETS.wic.name,
+        "TODO: tax_1040",
+    ],
+)
+def test_generate_dataset_with_state_filtered(dataset_name: str, request, mocker, tmpdir):
+    """Test that values returned by dataset generators are only for the specified state"""
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    dataset = DATASETS.get_dataset(dataset_name)
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+
+    # Skip noising (noising can incorrect select another state)
+    mocker.patch("pseudopeople.interface.noise_dataset", side_effect=_mock_noise_dataset)
+
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir, state=STATE)
+
+    assert (noised_data[dataset.state_column] == STATE).all()
+
+
+@pytest.mark.parametrize(
+    "dataset_name",
+    [
+        DATASETS.census.name,
+        DATASETS.acs.name,
+        DATASETS.cps.name,
+        DATASETS.tax_w2_1099.name,
+        DATASETS.wic.name,
+        "TODO: tax_1040",
+    ],
+)
+def test_generate_dataset_with_state_unfiltered(dataset_name: str, request, mocker, tmpdir):
+    """Test that values returned by dataset generators are for all locations if state unspecified"""
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    dataset = DATASETS.get_dataset(dataset_name)
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+
+    # Skip noising (noising can incorrect select another state)
+    mocker.patch("pseudopeople.interface.noise_dataset", side_effect=_mock_noise_dataset)
+
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir)
+
+    assert (noised_data[dataset.state_column] == STATE).any()
+    assert (noised_data[dataset.state_column] != STATE).any()
```

Comment:
Nit: maybe
```
assert len(noised_data[dataset.state_column].unique()) > 1
```
Dunno which is easier to understand, but this only iterates over the data once.

Reply:
Nice, good suggestion.